fenix/app/src/test/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckNavigationMiddlewareTest.kt

```diff
@@ -0,0 +1,73 @@
+package org.mozilla.fenix.shopping.middleware
+
+import mozilla.components.support.test.ext.joinBlocking
+import mozilla.components.support.test.libstate.ext.waitUntilIdle
+import mozilla.components.support.test.rule.MainCoroutineRule
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+import org.junit.Test
+import org.mozilla.fenix.shopping.store.ReviewQualityCheckAction
+import org.mozilla.fenix.shopping.store.ReviewQualityCheckState
+import org.mozilla.fenix.shopping.store.ReviewQualityCheckStore
+
+class ReviewQualityCheckNavigationMiddlewareTest {
+
+    @get:Rule
+    val coroutinesTestRule = MainCoroutineRule()
+    private val dispatcher = coroutinesTestRule.testDispatcher
+    private val scope = coroutinesTestRule.scope
+
+    @Test
+    fun `WHEN opening an external link THEN the link should be opened in a new tab`() {
+        var isOpenedInSelectedTab = false
+        var isOpenedInNewTab = false
+        val store = ReviewQualityCheckStore(
+            middleware = listOf(
+                ReviewQualityCheckNavigationMiddleware(
+                    openLinkInNewTab = {
+                        isOpenedInNewTab = true
+                    },
+                    openLinkInSelectedTab = {
+                        isOpenedInSelectedTab = true
+                    },
+                    scope = scope,
+                ),
+            ),
+        )
+        store.waitUntilIdle()
+        dispatcher.scheduler.advanceUntilIdle()
+
+        store.dispatch(ReviewQualityCheckAction.OpenLink(ReviewQualityCheckState.LinkType.ExternalLink("www.fakespot.com"))).joinBlocking()
```

Comment:
Thanks for reviewing this! The action should invoke the callback and change the `isOpenedInNewTab` variable, no network connection is created. I'll replace the placeholder link to avoid any confusion.